src/rules/unit-blacklist/__tests__/index.js

```diff
@@ -0,0 +1,47 @@
+import {
+  ruleTester,
+  warningFreeBasics,
+} from "../../../testUtils"
+import rule, { ruleName, messages } from ".."
+
+const testRule = ruleTester(rule, ruleName)
+
+testRule([
+  "px",
+  "vmin",
+], tr => {
+  warningFreeBasics(tr)
+
+  tr.ok("a { line-height: 1; }")
+  tr.ok("a { color: #000; }")
+  tr.ok("a { top: 0; left: 0; }")
+  tr.ok("a { font-size: 100%; }")
+  tr.ok("a { line-height: 1.2rem; }")
+
+  tr.ok("/* 100px */", "ignore unit within comments")
+  tr.ok("a::before { content: \"10px\"}", "ignore unit within quotes")
+
+  tr.ok("a { font-size: $fs10px; }", "ignore preprocessor variable includes unit")
+  tr.ok("a { font-size: --some-fs-10px; }", "ignore css variable includes unit")
+
```

Comment:
Can we have a test for a _shorthand property_, one for a _value-list_, one for _functions_ please and a final one for _nested functions_? e.g.

```css
a { margin: 0 10em 5rem 2in; }
```

```css
a { background-position: top right, 1em 5vh; }
```

```css
a { top: calc(10em - 3em); }
```

```css
a { background-image: linear-gradient(to right, white calc(100% - 50em), silver); }
```